onnxruntime/core/providers/armnn/math/gemm.h

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) 2020, NXP Semiconductor, Inc. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/framework/op_kernel.h"
+#include "core/util/math.h"
+#include "core/util/math_cpuonly.h"
+#include "core/providers/cpu/math/gemm.h"
+#include "core/providers/armnn/armnn_execution_provider.h"
+
+namespace onnxruntime {
+namespace armnn_ep {
+
+typedef std::map<OpKernel*, armnn::NetworkId>::iterator GEMMLayersIterator;
+
+template <typename T>
+class Gemm : public onnxruntime::Gemm<T> {
+ public:
+  Gemm(const OpKernelInfo& info) : onnxruntime::Gemm<T>(info) {
+    int64_t temp;
+
+    ORT_ENFORCE(info.GetAttr<int64_t>("transA", &temp).IsOK());
+    trans_A_ = temp == 0 ? CblasNoTrans : CblasTrans;
+    ORT_ENFORCE(info.GetAttr<int64_t>("transB", &temp).IsOK());
+    trans_B_ = temp == 0 ? CblasNoTrans : CblasTrans;
+
+    ORT_ENFORCE(info.GetAttr<float>("alpha", &alpha_).IsOK());
+    ORT_ENFORCE(info.GetAttr<float>("beta", &beta_).IsOK());
+  }
+
+  Status Compute(OpKernelContext* context) const override {
```

Comment:
Just a question: I find this a bit intriguing, shouldn't we just have the declaration in .h file and define the member function/constructor in .cc file(we can make them inline)? I see this in quite a few places. Is there any advantage of doing this in general?

Reply:
The gemm operation is the only one with this implementation, as far as I know. It was done this way because we started from the cpu model. It can be changed.